share/admin.py

```diff
@@ -353,8 +355,58 @@ class ShareUserAdmin(admin.ModelAdmin):
     inlines = (SourceAdminInline,)
 
 
+class SourceAddForm(forms.ModelForm):
+
+    class Meta:
+        model = Source
+        exclude = ('user', 'long_title', )
+        widgets = {'name': TextInput}
+
+    def validate_unique(self):
+        # Forces validation checks on every field
+        try:
+            self.instance.validate_unique()
```

Comment:
There isn't a field on the form for `long_title` though so it just blows up. Is that intended?